src/main/java/sorald/Repair.java

```diff
@@ -65,7 +69,13 @@ public void repair() {
             final Path inputDir = inOutPaths.getLeft();
             final Path outputDir = inOutPaths.getRight();
 
-            SoraldAbstractProcessor<?> processor = createProcessor(ruleKey);
+            Set<RuleViolation> ruleViolations =
+                    ProjectScanner.scanProject(
+                            inputDir.toFile(),
+                            FileUtils.getClosestDirectory(inputDir.toFile()),
+                            Checks.getCheckInstance(Integer.toString(ruleKey)));
+            SoraldAbstractProcessor<?> processor =
+                    createProcessor(ruleKey).setRuleViolations(ruleViolations);
```

Comment:
We now set rule violations here, at the very top of the repair process. This makes it easy to swap these currently discovered violations for something we construct ourselves from parsing rule IDs for targeted repair.

Reply:
Nice!